core/tx_pool.go

```diff
@@ -407,10 +418,16 @@ func (pool *TxPool) Add(tx *types.Transaction) error {
 	pool.mu.Lock()
 	defer pool.mu.Unlock()
 
+	state, err := pool.currentState()
+	if err != nil {
+		return err
+	}
```

Comment:
Please move this down to right before the `promoteExecutables` method call. It looks a bit strange that we retrieve the state here, then do something else and only afterwards use the state. Or is there a particular reason for this being before `pool.add`?